1_fetch/src/cross_validation_functions.R

```diff
@@ -0,0 +1,21 @@
+
+get_nested_gages<-function(gagesii,nav_distance_km){
+  gage_IDs <- as.character(gagesii$ID)
+  COMIDs <- as.character(gagesii$COMID)
+  
+   #assigns a 1 if the row name gage is upstream of the column name gage
+  upstream_df <- data.frame(matrix(ncol = length(gage_IDs), nrow = length(gage_IDs)))
+  colnames(upstream_df) <- gage_IDs; rownames(upstream_df) <- gage_IDs
+  upstream_df[,]<-0
+  for (i in 1:length(gage_IDs)){
+    flowline <- navigate_nldi(list(featureSource = "comid", 
+                                       featureID = COMIDs[i]), 
+                                  mode = "upstreamTributaries",
+                                  distance_km=nav_distance_km)
+    upstream_list <- flowline$UT_flowlines$nhdplus_comid
+    upstream_df[which(COMIDs %in% upstream_list), i] <- 1
+    message(gage_IDs[i])
+  }
```

Comment:
It would be helpful to report other information about nestedness to help us choose a threshold for how nested is too much for gages to be in a separate group. Does nhdplusTools have a function for pulling the upstream catchment area? Reporting the proportion overlap would be helpful. The scripts I sent you from the Drought project might do some of this because theirs is based on area

Reply:
I can take a look at that script and see what nhdplusTools has but I wonder if it might be easier to just use dataRetreival to get the drainage areas. Then, instead of just a 1/0 in the matrix, I could put the drainage area ratio upstream/downstream in the matrix instead. Let me know what you think of that idea.

Reply:
oh that's definitely going to be faster! Good idea